x-pack/plugins/security/server/audit/audit_logger.ts

```diff
@@ -7,7 +7,7 @@
 import { LegacyAPI } from '../plugin';
 
 export class SecurityAuditLogger {
-  constructor(private readonly auditLogger: LegacyAPI['auditLogger']) {}
+  constructor(private readonly getAuditLogger: () => LegacyAPI['auditLogger']) {}
```

Comment:
__note:__ made this to be similar with ESO plugin and also be able to instantiate this logger (and hence initialize Saved Objects wrappers) outside of `registerLegacyAPI` method.